api/.squawk.toml

```
@@ -0,0 +1,2 @@
+pg_version = "12.11"
+excluded_rules = ["ban-drop-column", "prefer-identity", "prefer-timestamptz"]
```

```suggestion
excluded_rules = [
    "ban-drop-column",
    "prefer-identity",
    "prefer-timestamptz",
]
```
Est-ce que l'on peut commenter/documenter pourquoi ces 3 règles sont (pour le moment) exclues ? En particulier `prefer-identity`

---

- `ban-drop-column` : nous sommes les seuls clients à utiliser notre base de donnée, donc on devrait pouvoir nettoyer nos tables et avoir des breaking changes
- `prefer-identity` : cette règle n'aime pas avoir une sequence comme clé primaire, mais c'est le comportement par défaut de sqlalchemy et toutes nos tables ont une clé primaire séquentielle
- `prefer-timestamptz` : on stocke tout en UTC, ce qui est une bonne pratique d'internationalisation

Ces explications te vont ?

---

En lançant Squawk sur toutes nos migrations, j'ai ce message : 

```
$ alembic upgrade f0fbdcf9961d:post@head --sql | ../node_modules/squawk-cli/js/bin/squawk --config .squawk.toml
[...]
stdin:86:2: warning: ban-drop-not-null

  86 | -- Running upgrade fadf7f6f4c60 -> 2cc0a6822cf8
  87 | 
  88 | ALTER TABLE cashflow ALTER COLUMN "reimbursementPointId" DROP NOT NULL;

  note: Dropping a NOT NULL constraint may break existing clients.
```

Ici, pour cette migration, c'est un besoin légitime. Doit-on ignorer l'erreur `ban-drop-not-null` globalement (dans le fichier de configuration de Squawk), ou peut-on l'ignorer unitairement dans cette migration-là seulement ?

---

Je pense qu'on peut ignorer globalement `ban-drop-not-null` puisque nous sommes les seuls clients de notre base de donnée